modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -221,15 +222,21 @@ protected void doRun() throws Exception {
                 }
 
                 @Override
-                public void onFailure(Exception t) {
-                    if (t instanceof ResponseException) {
-                        ResponseException re = (ResponseException) t;
+                public void onFailure(Exception e) {
+                    if (e instanceof ResponseException) {
+                        ResponseException re = (ResponseException) e;
                         if (RestStatus.TOO_MANY_REQUESTS.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
-                            listener.onRetryableFailure(t);
+                            listener.onRetryableFailure(e);
                             return;
                         }
+                        if (RestStatus.UNAUTHORIZED.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
```

Comment:
I can't just pass on the response code in the general case because only `ElasticsearchSecurityException` supports settings the response status. Other exceptions figure out their own status.